lib/Backend/arm64/EncoderMD.cpp

```diff
@@ -551,11 +551,20 @@ int EncoderMD::EmitMovConstant(Arm64CodeEmitter &Emitter, IR::Instr *instr, _Emi
         Assert(src1->IsLabelOpnd());
         IR::LabelInstr* labelInstr = src1->AsLabelOpnd()->GetLabel();
 
-        // Here the LabelOpnd's offset is a post-lower immediate value; we need
-        // to mask in just the part indicated by our own shift amount, and send
-        // that along as our immediate load value.
-        uintptr_t fullvalue = labelInstr->GetOffset();
-        immediate = (fullvalue & (0xffff << shift)) >> shift;
+        if (labelInstr->m_isDataLabel)
+        {
+            // If the label is a data label, we don't know the label address yet so handle it as a reloc.
+            EncodeReloc::New(&m_relocList, RelocTypeLabelImmed, m_pc, labelInstr, m_encoder->m_tempAlloc);
+            immediate = 0;
+        }
+        else
+        {
+            // Here the LabelOpnd's offset is a post-lower immediate value; we need
+            // to mask in just the part indicated by our own shift amount, and send
+            // that along as our immediate load value.
+            uintptr_t fullvalue = labelInstr->GetOffset();
+            immediate = (fullvalue & (0xffff << shift)) >> shift;
+        }
```

Comment:
It is simpler and clearer (IMO) to write this as immediate = (fullvalue >> shift) & 0xffff